### ISSUE TITLE
S1025 - False positive when escaping a string

### ISSUE THREAD

**Issue by vectorman1** [go-critic/go-critic#1222]
I have the following code snippet:

```go
func EscapeString(s string) string {
	return fmt.Sprintf(`%s`, s)
}
```

This gives me a false positive as per the `S1025` rule. What I am achieving here is escaping the string passed in, and I think this is a valid usecase. If there are other ways to accomplish this, let me know, however, I do still think this is an issue.

Thank you

**Comment by quasilyte** [comment]
What is a S1025 rule?
Are you sure it's not a staticcheck rule?

Also, how exactly the code above escapes the string? Maybe `%q` format was intended?
`%s` does not escaping AFAIK.

**Comment by vectorman1** [comment]
[Link to the rule](https://staticcheck.io/docs/checks/#S1025)

The escape works due to the backticks used on the string. I was surprised as well :)

**Comment by dominikh** [comment]
It's also not a false positive, and the backtick has zero effect on the way fmt.Sprintf works or interprets the %s verb.
